lib/catalog/approval_transition.rb

```diff
@@ -39,28 +37,29 @@ def state_transitions
 
     def submit_order
       finalize_order
-      @order_item.update_message("info", "Submitting Order #{@order_item.order_id} for provisioning ")
+      @order_item.order.update_message("info", "Submitting Order for provisioning")
       Catalog::SubmitNextOrderItem.new(@order_item.order_id).process
     rescue ::Catalog::TopologyError => e
       Rails.logger.error("Error Submitting Order #{@order_item.order_id}, #{e.message}")
-      @order_item.update_message("error", "Error Submitting Order #{@order_item.order_id}, #{e.message}")
+      @order_item.order.update_message("error", "Error Submitting Order: #{e.message}")
+      @order_item.mark_failed("Error Submitting Order: #{e.message}")
```

Comment:
It is still not correct because it is not `@order_item` that is failing but the order's first orderable item is failing. Maybe we should catch topo error in `SubmitNextOrderItem` and `mark_failed` there.